File: Week3/Thursday/pad-array/python/pad_array.py
def pad(array, min_size, value = None):
    """
    Pad will accept a list, a minimum size (non-negative integer) for the list, and an optional argument of what the list should be "padded" with.

    If the list is longer than what the minimun size is, then continue to add the padded value to the end till the minimum size of the list is reached

    If the list's length is less than the minimum size, `pad` should return a new list padded with the pad value up to the minimum size.

    >>>> pad([1,2,3], 5)

    returns >>> [1,2,3,None,None]

    """
    results = array
    if len(array) < min_size:
        size_difference = min_size - len(array)
        for i in range(size_difference):
            results.append(value)
    return results

File: Week3/Thursday/pad-array/js/padArray.js


/** 
 * If length of the array is > minSize then
 * take the difference between the length and minSize and push on that many of the value
 * If the length of array is <= the min size then
 * return array
 */
const pad = (array, minSize, value = null) => {
  let results = array
  if (array.length < minSize) {
    sizeDifference = minSize - array.length
    for (let i = 0; sizeDifference > i; i++){
      results.push(value)
    }
  }
  return results
}

// console.log(pad([1,2,3], 5))

module.exports = {
  pad
}